NO-JIRA: certregenerationcontroller: Improve goroutine mgmt

tchap opened the pull request:
Also use typed queues, the untyped are deprecated.

This is just refactoring and handling some todos.
I took the same approach as implemented in core [k8s controllers](https://github.com/kubernetes/kubernetes/blob/5a7c90fb7679071d3d4d6708ae56048795791aaa/pkg/controller/job/job_controller.go#L245).

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **Refactor**
  * Modernized controller infrastructure to a shared factory-based controller for simpler reconciliation and informer wiring.
  * Replaced ad‑hoc queue/workers with standardized lifecycle handling, removing custom run/worker/queue management.
  * Replaced unmanaged goroutines with tracked worker lifecycle and coordinated shutdown to improve startup/shutdown reliability.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

p0lyn0mial commented on pkg/cmd/certregenerationcontroller/cabundlesyncer.go:
```diff
@@ -74,23 +80,24 @@ func NewCABundleController(
 }
 
 func (c *CABundleController) Run(ctx context.Context) {
```
could this be rewritten to use standard ` factory.New()` controller ?

tchap replied:
Rewritten to use `factory.New`.

tchap committed:
certregenerationcontroller: Improve goroutine mgmt

Use a WaitGroup to manage all goroutines.

tchap committed:
certregenerationcontroller: Use factory.New

Rewrite the controller init to use factory.New to remove unnecessary
boilerplate code.

tchap commented on pkg/cmd/certregenerationcontroller/cmd.go:
```diff
@@ -117,8 +118,17 @@ func (o *Options) Run(ctx context.Context, clock clock.Clock) error {
 
-	go configInformers.Start(ctx.Done())
-	go featureGateAccessor.Run(ctx)
+	var wg sync.WaitGroup
+	defer wg.Wait()
+	// cancel must happen before wg.Wait (so in a later defer), otherwise we can get stuck on early return.
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	configInformers.Start(ctx.Done())
+
+	wg.Go(func() {
+		featureGateAccessor.Run(ctx)
```
Added.

tchap commented:
/retest
